### ISSUE TITLE
Unable to install Microsoft.Azure.Devices in a Xamarin Forms android project

### ISSUE THREAD

**Issue by albaek75** [Azure/azure-iot-sdk-csharp#106]
Hi

I am getting this error when trying to install Microsoft.Azure.Devices to Xamarin Forms android project.

Error Could not install package 'Microsoft.Azure.Devices 1.2.4'. You are trying to install this package into a project that targets 'MonoAndroid,Version=v7.1', but the package does not contain any assembly references or content files that are compatible with that framework. For more information, contact the package author.

Is there any plans to add this support?

**Comment by tameraw** [comment]
@albaek75 - [Microsoft.Azure.Devices 1.2.6](https://www.nuget.org/packages/Microsoft.Azure.Devices/) is our first package that supports .NetStandard (version 1.3 is what we have).
Please try with this package and let us know if still having issues.

**Comment by sarsijravi** [comment]
@tameraw -
Is this only for MonoAndroid? Beacause I tried installing for the **portable class in Xamarin forms** and got the following error:
 **Could not install package 'Microsoft.Azure.Devices 1.2.6'. You are trying to install this package into a project that targets 
'.NETPortable,Version=v4.5,Profile=Profile259'**

Any Solutions?

**Comment by tameraw** [comment]
@sarsijravi - Please try with a Profile 7 and let us know if still having issues.

**Comment by sarsijravi** [comment]
@tameraw 
Already tried. Doesn't work.
Got the following error again:
**Could not install package **'Microsoft.Azure.Devices 1.2.7'**. You are trying to install this package into a project that targets '.**NETPortable,Version=v4.5,Profile=Profile7',** but the package does not contain any assembly references or content files that are compatible with that framework**

**Comment by tameraw** [comment]
@albaek75 - we'll investigate the issue. Thanks.

**Comment by ryan-z-johnson** [comment]
@sarsijravi sorry for the long delay. I have been able to reproduce your issue. The way we are encouraging people who are using a mono based platform is to create a .NetStandard 1.3 class library and add the nugget package to that, and reference the library in your Xamarian.Android application. Here is a GitHub Repo with the structure I am talking about. https://github.com/ryan-z-johnson/AzureIoTXamarin

If you have any more questions please let me know.

**Comment by CIPop** [comment]
We're tracking multiple issues caused by the PCL libraries. The decision was to remove the PCL binary from our next release.
Please follow the above workaround until we complete this work.

Tentatively closing this as a duplicate of #138 .

**Comment by coltonf93** [comment]
@ryan-z-johnson 
This workaround successfully builds the Android application with the device client dependency but breaks when you actually try to use it.
e.g. add this constructor to the SharedLibrary class1 file and try building the android project.
```
public Class1()
{
    var _deviceClient = DeviceClient.Create("", new DeviceAuthenticationWithRegistrySymmetricKey("", ""), TransportType.Mqtt);
}
```
It throws:
`Exception while loading assemblies: System.IO.FileNotFoundException: Could not load assembly 'Microsoft.Azure.Devices.Client, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. Perhaps it doesn't exist in the Mono for Android profile?
`	
Any ideas?

**Comment by CIPop** [comment]
@coltonf93 I've seen issues when either Xamarin or Visual Studio are not up to date. 
Also, can you confirm you're using the latest release of our SDK?

**Comment by coltonf93** [comment]
I'm building the example project/workaround found [here](https://github.com/ryan-z-johnson/AzureIoTXamarin). 
Everything should be up to date ([Versions Used](https://jpst.it/1bkpE)).

**Comment by CIPop** [comment]
@coltonf93 We're not supporting PCL at all in either LTS or master.
This issue was due to the fact that we were shipping a PCL target within the NuGet package. The last releases do not have such PCL target.

After upgrading VS and Xamarin, you will need to also switch to the latest Xamarin.Forms template:
![image](https://user-images.githubusercontent.com/8635911/36695471-4e79121c-1af6-11e8-9606-f5c5ccbd7c67.png)

Select the ".NET Standard" Code Sharing Strategy:
![image](https://user-images.githubusercontent.com/8635911/36695503-5fa304f8-1af6-11e8-9ceb-42319350f08e.png)

Reference the latest non-preview [Azure SDK DeviceClient](https://www.nuget.org/packages/Microsoft.Azure.Devices.Client/1.6.2) within the shared project.
You should not use any of the workarounds presented here as they are not applicable to the new SDK versions.

A Debug build should work for Android.
Release builds for Android are still blocked by multiple issues within the Xamarin platform. Please use https://github.com/azure/azure-iot-sdk-csharp/projects/1 to track these issues.

If this doesn't work, please create a new issue with the details.

**Comment by coltonf93** [comment]
I followed your recommendation, and it leads to [this exact issue](https://github.com/Azure/azure-iot-sdk-csharp/issues/373). I'll just follow up there.

**Comment by CIPop** [comment]
Thanks @coltonf93 ! Let's continue in #373.
I'm currently working on adding netstandard2.0 and removing the UAP builds to simplify the investigation and pick up more dependencies not available in netstandard1.3. Hopefully that will solve some of the Xamarin and UWP issues caused by some of the dependencies.

**Comment by MujasSam** [comment]
Severity	Code	Description	Project	File	Line	Suppression State
Error		Exception while loading assemblies: System.IO.FileNotFoundException: Could not load assembly 'Microsoft.Azure.Devices.Shared.PCL, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. Perhaps it doesn't exist in the Mono for Android profile?
File name: 'Microsoft.Azure.Devices.Shared.PCL.dll'
  




@CIPop 

I am facing the same issue. Any solution for this. **(Microsoft.Azure.Devices.Client.PCL)**

 at Java.Interop.Tools.Cecil.DirectoryAssemblyResolver.Resolve(AssemblyNameReference reference, ReaderParameters parameters)
   at Xamarin.Android.Tasks.ResolveAssemblies.AddAssemblyReferences(DirectoryAssemblyResolver resolver, ICollection`1 assemblies, AssemblyDefinition assembly, Boolean topLevel)
   at Xamarin.Android.Tasks.ResolveAssemblies.Execute(DirectoryAssemblyResolver resolver)	BlueBox.Android	C:\Program Files (x86)\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets	1413

**Comment by CIPop** [comment]
@MujasSam We don't support the PCL binary anymore.
Please try with 1.7.0 which added netstandard2.0 support.
